src/renderer/domains/network/multisig-operation/service.ts

```diff
@@ -129,10 +129,28 @@ const mergeEvents = (oldEvents: MultisigEvent[], events: MultisigEvent[]) =>
     sort: (a, b) => a.blockCreated - b.blockCreated,
   });
 
-const mergeMultisigOperations = (a: MultisigOperation[], b: MultisigOperation[]): MultisigOperation[] => {
+const mergeMultisigOperations = (
+  oldOperations: MultisigOperation[],
+  operations: MultisigOperation[],
+): MultisigOperation[] => {
+  /**
+   * Because we work with Events AT BEST, it may happen that our db has extra
+   * operations that are not on the chain anymore nor were executed. We want to
+   * filter them out.
+   *
+   * For example, if we have a fork and operation was executed in the different
+   * block than we first recevied event of.
+   */
+  const oldFilteredForPending = oldOperations.filter(
```

Comment:
Looks like you're removing old operations for chains and accounts that are existed in new operations. let's put it that way in code (like what is `oldFilteredForPending`?) and optimize it. Now it's terribly unoptimized